### ISSUE TITLE
Alter config at run time?

### ISSUE THREAD

**Issue by jdtsmith** [ipython/ipython#12990]
I'd like to do the equivalent of config option

```python
c.InteractiveShell.display_page = True
```

at run-time (for an Emacs front-end to iPython I'm building).  I've found that 

```python
get_ipython().config.InteractiveShell.display_page
``` 

shows this info, but setting it doesn't disable the pager.

**Comment by Carreau** [comment]
Try `%config ZMQInteractiveShell.display_page=True` ?

`%config` on its own will show you which instances you can configure, and you need `ZMQInteractiveShell` as you want to target the actual type of the instance instead of it's superclass when changing config at runtime. 

`get_ipython().display_page=True` might also work, as the only instance of a InteractiveShell is returned by `get_ipython()` though using `%config ....` may do some extra stuff, i'm not sure. 

get_ipython().config.InteractiveShell.display_page will change the config object which will likely only affect subsequetnly created objects; so if you nest IPython shells or start the debugger.

**Comment by jdtsmith** [comment]
Thanks.  Unfortunately these don't seem to do the trick.   I have a `%config TerminalInteractiveShell.display_page` but no `ZMQ` version.  The former does tie to the config input `c.InteractiveShell.display_page`, but the setting at iPython start stays in effect.  Example:

```
In [1]: %config TerminalInteractiveShell.display_page
Out[1]: False

In [3]: import numpy

In [4]: %pdoc numpy # pipes through a pager

%config TerminalInteractiveShell.display_page = True

In [6]: %config TerminalInteractiveShell.display_page
Out[6]: True

In [7]: %pdoc numpy # still pages :(

In [13]: get_ipython().display_page = True

In [14]: %pdoc numpy # Still pages

In [15]: get_ipython().display_page
Out[15]: True
```

My guess is that IPython sets up an output (paged or not) for commands like `%pdoc` at startup, such that any later config is ineffective.  Any thoughts on whether this is true, or how to ask it to reconfigure the potentially paged output?

**Comment by Carreau** [comment]
does the following do the trick ?

```python
from IPython.core import page
get_ipython().set_hook('show_in_pager', page.as_hook(page.display_page), 90)
```

**Comment by Carreau** [comment]
Yes, this is indeed a hook priority. I'll see if I can hook up the config
to update the hooks when it changes.
